source/fe/mapping_q_generic.cc

```diff
@@ -1148,27 +1148,113 @@ namespace internal
 
 
 
+      /**
+       * Using the given 1D polynomial basis and the position of the mapping
+       * support points, compute the mapped location of that point in real
+       * space. This function is much faster than the other implementation
+       * going via the expanded shape functions in InternalData because it
+       * directly works in the tensor product form. This also gives the
+       * derivative almost for free (less than 2x the cost of simply the
+       * values), so we always compute it.
+       */
+      template <int dim, int spacedim>
+      std::pair<Point<spacedim>, Tensor<2, spacedim>>
+      compute_mapped_location_of_point(
+        const std::vector<Point<spacedim>> &                points,
+        const std::vector<Polynomials::Polynomial<double>> &poly,
+        const std::vector<unsigned int> &                   renumber,
+        const Point<dim> &                                  p)
```

Comment:
This is the central function that implements the fast evaluation of the `FE_Q` interpolation in a point (including its derivative). This could be generalized for all evaluations of tensor product interpolations if we find an appropriate function interface as to be discussed in #10968, so the implementation could move somewhere else once we found a strategy.